```
FAILED tests/components/netatmo/test_camera.py::test_camera_image_local[pyloop]
FAILED tests/components/netatmo/test_camera.py::test_camera_image_vpn[pyloop]
FAILED tests/components/netatmo/test_camera.py::test_camera_image_raises_exception[pyloop]
```

unrelated test issue

--- a/homeassistant/components/flux_led/light.py
+++ b/homeassistant/components/flux_led/light.py
@@ -43,7 +43,6 @@ from homeassistant.const import (
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers import device_registry as dr, entity_platform
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
@@ -278,21 +277,15 @@ class FluxLight(CoordinatorEntity, LightEntity):
         self._custom_effect_colors = custom_effect_colors
         self._custom_effect_speed_pct = custom_effect_speed_pct
         self._custom_effect_transition = custom_effect_transition
+        old_protocol = self._bulb.protocol == "LEDENET_ORIGINAL"
         if self.unique_id:
-            device_info: DeviceInfo = {
+            self._attr_device_info = {
                 "connections": {(dr.CONNECTION_NETWORK_MAC, self.unique_id)},
+                ATTR_MODEL: f"0x{self._bulb.raw_state[1]:02X}",
+                ATTR_SW_VERSION: "1" if old_protocol else str(self._bulb.raw_state[10]),
                 ATTR_NAME: self.name,
                 ATTR_MANUFACTURER: "FluxLED/Magic Home",
             }
-            if self._bulb.raw_state:
-                device_info[ATTR_MODEL] = f"0x{self._bulb.raw_state[1]:02X}"
-            else:
-                device_info[ATTR_MODEL] = "LED Lights"
-            if self._bulb.protocol != "LEDENET_ORIGINAL" and self._bulb.raw_state:
-                device_info[ATTR_SW_VERSION] = str(self._bulb.raw_state[10])
-            else:
-                device_info[ATTR_SW_VERSION] = "1"
-            self._attr_device_info = device_info
 
     @property
     def is_on(self) -> bool:

--- a/tests/components/flux_led/test_light.py
+++ b/tests/components/flux_led/test_light.py
@@ -93,8 +93,13 @@ async def test_light_no_unique_id(hass: HomeAssistant) -> None:
     assert state.state == STATE_ON
 
 
-async def test_light_firmware_old_protocol(hass: HomeAssistant) -> None:
-    """Test a light firmware old protocol."""
+@pytest.mark.parametrize(
+    "protocol,sw_version,model", [("LEDENET_ORIGINAL", 1, 0x35), ("LEDENET", 8, 0x33)]
+)
+async def test_light_device_registry(
+    hass: HomeAssistant, protocol: str, sw_version: int, model: int
+) -> None:
+    """Test a light device registry entry."""
     config_entry = MockConfigEntry(
         domain=DOMAIN,
         data={CONF_HOST: IP_ADDRESS, CONF_NAME: DEFAULT_ENTRY_TITLE},
@@ -102,8 +107,9 @@ async def test_light_firmware_old_protocol(hass: HomeAssistant) -> None:
     )
     config_entry.add_to_hass(hass)
     bulb = _mocked_bulb()
-    bulb.protocol = "LEDENET_ORIGINAL"
-    bulb.raw_state[1] = 0x35
+    bulb.protocol = protocol
+    bulb.raw_state[1] = model
+    bulb.raw_state[10] = sw_version
     with _patch_discovery(no_device=True), _patch_wifibulb(device=bulb):
         await async_setup_component(hass, flux_led.DOMAIN, {flux_led.DOMAIN: {}})
         await hass.async_block_till_done()
@@ -112,31 +118,8 @@ async def test_light_firmware_old_protocol(hass: HomeAssistant) -> None:
     device = device_registry.async_get_device(
         identifiers={}, connections={(dr.CONNECTION_NETWORK_MAC, MAC_ADDRESS)}
     )
-    assert device.sw_version == "1"
-    assert device.model == "0x35"
-
-
-async def test_light_firmware_new_protocol(hass: HomeAssistant) -> None:
-    """Test a light firmware new protocol."""
-    config_entry = MockConfigEntry(
-        domain=DOMAIN,
-        data={CONF_HOST: IP_ADDRESS, CONF_NAME: DEFAULT_ENTRY_TITLE},
-        unique_id=MAC_ADDRESS,
-    )
-    config_entry.add_to_hass(hass)
-    bulb = _mocked_bulb()
-    bulb.raw_state[10] = 8
-    bulb.raw_state[1] = 0x33
-    with _patch_discovery(no_device=True), _patch_wifibulb(device=bulb):
-        await async_setup_component(hass, flux_led.DOMAIN, {flux_led.DOMAIN: {}})
-        await hass.async_block_till_done()
-
-    device_registry = dr.async_get(hass)
-    device = device_registry.async_get_device(
-        identifiers={}, connections={(dr.CONNECTION_NETWORK_MAC, MAC_ADDRESS)}
-    )
-    assert device.sw_version == "8"
-    assert device.model == "0x33"
+    assert device.sw_version == str(sw_version)
+    assert device.model == f"0x{model:02X}"
 
 
 async def test_rgb_light(hass: HomeAssistant) -> None:

--- a/tests/components/flux_led/test_light.py
+++ b/tests/components/flux_led/test_light.py
@@ -103,6 +103,7 @@ async def test_light_firmware_old_protocol(hass: HomeAssistant) -> None:
     config_entry.add_to_hass(hass)
     bulb = _mocked_bulb()
     bulb.protocol = "LEDENET_ORIGINAL"
+    bulb.raw_state[1] = 0x35
     with _patch_discovery(no_device=True), _patch_wifibulb(device=bulb):
         await async_setup_component(hass, flux_led.DOMAIN, {flux_led.DOMAIN: {}})
         await hass.async_block_till_done()
@@ -112,6 +113,7 @@ async def test_light_firmware_old_protocol(hass: HomeAssistant) -> None:
         identifiers={}, connections={(dr.CONNECTION_NETWORK_MAC, MAC_ADDRESS)}
     )
     assert device.sw_version == "1"
+    assert device.model == "0x35"
 
 
 async def test_light_firmware_new_protocol(hass: HomeAssistant) -> None:
@@ -124,6 +126,7 @@ async def test_light_firmware_new_protocol(hass: HomeAssistant) -> None:
     config_entry.add_to_hass(hass)
     bulb = _mocked_bulb()
     bulb.raw_state[10] = 8
+    bulb.raw_state[1] = 0x33
     with _patch_discovery(no_device=True), _patch_wifibulb(device=bulb):
         await async_setup_component(hass, flux_led.DOMAIN, {flux_led.DOMAIN: {}})
         await hass.async_block_till_done()
@@ -133,6 +136,7 @@ async def test_light_firmware_new_protocol(hass: HomeAssistant) -> None:
         identifiers={}, connections={(dr.CONNECTION_NETWORK_MAC, MAC_ADDRESS)}
     )
     assert device.sw_version == "8"
+    assert device.model == "0x33"
 
 
 async def test_rgb_light(hass: HomeAssistant) -> None:

